enhancements/olm-admission-webhooks.md

```diff
@@ -0,0 +1,137 @@
+	SideEffects             *admissionregistrationv1.SideEffectClass        `json:"sideEffects"`
+	TimeoutSeconds          *int32                                          `json:"timeoutSeconds,omitempty"`
+	AdmissionReviewVersions []string                                        `json:"admissionReviewVersions"`
+	ReinvocationPolicy      *admissionregistrationv1.ReinvocationPolicyType `json:"reinvocationPolicy,omitempty"`
+}
+
+// WebhookDescription is used to provide OLM with information about the Webhook being shipped with the operator
+type WebhookDescription struct {
+	// Type is used to identify the admission webhook as a validating or mutating webhook. Accepted values include `validating` and `mutating`
+	Type                    WebhookAdmissionType                            `json:"type"`
+	// DeploymentName must match one of the deployments included in the CSV.
+	DeploymentName          string                                          `json:"deploymentName,omitempty"`
+	// ContainerPort specifies which port the service should forward requests to.
+	ContainerPort           int32                                           `json:"containerPort,omitempty"`
+	// Webhook contains the information that must be sent to the validating or mutating webhook. 
+	Webhook                 Webhook                                      `json:"webhook"`
+}
+```
+
+It important to note that the Webhook structure does not provide users with the ability to set the WebhookClientConfig and NamespaceSelctor fields. Both of these fields will be generated via OLM. In detail:
+
+- OLM will create the WebhookConfigClient after generating the Service and the CA-Cert.
+- The NamespaceSelector field will be generated based on the OperatorGroup that the operator is deployed in. When creating the OperatorGroup, OLM will apply a label to all namespaces that are included in that operator group. Then, when creating the Webhook, OLM will update the namespace selector to use that label.
+
+### Securing webhook via certificates
+
+The certificate creation and lifecycle management is planned to utilize the same certs package that is already in OLM managing the certificates for api services. OLM has tested this feature extensively at this point and could easily port it to webhooks.
+
+### Integration with Cert Management tools
+
+There is a great deal of community support for integration with certificate management projects such as the [service-ca operator](https://github.com/openshift/service-ca-operator) or [cert-manager](https://github.com/jetstack/cert-manager). Cluster Admins should be given the opportunity to integrate with either of these tools be disabling the OLM management of certs and adding appropriate annotations to the deployment and webhook resources via the [SubscriptionConfig resource](https://github.com/operator-framework/operator-lifecycle-manager/blob/8985872bfd5888c8d7d49e4dfc9be162a87691e7/pkg/api/apis/operators/v1alpha1/subscription_types.go#L41-L87).
```

Comment:
So for OpenShift downstream deployments we can assume the service-ca operator will be in the cluster. But for upstream, say we want to use cert-manager, can we assume that the certs will be generated as part of the webhook deployment? do we explicitly need to install cert-manager into the cluster and have it as an upstream dependency, or assume in upstream that the webhooks have `insecureSkipTLSCheck`?

Reply:
Good point, I had considered this but found that cert-manager isn't an OLM Managed operator. I had believed that by giving a Cluster Admin the ability to disable OLM Cert Management and specifying Annotations that the admin could integrate the operator with Cert-Manager from there. Otherwise, generic OLM Cert Management could be used.